kafka/verify_kafka_running.py

```diff
@@ -0,0 +1,159 @@
+        if os.WEXITSTATUS(os.system(command)) != os.EX_OK:
+            raise Exception('Create test topic unsuccessful')
+
+    def publish_messages(self):
+        """Publishes messages using worker 0 as the broker.
+
+           Fails if publishing is unsuccessful.
+
+        Raises:
+          Exception: if publishing results in KafkaError
+        """
+        node_name = self._get_address(WORKER_0_NODE_SUFFIX, PORT)
+        print('Publishing ' + node_name)
+
+        producer = KafkaProducer(
+            bootstrap_servers=[node_name],
+            retries=NUM_RETRIES)
+
+        try:
+            for i in range(NUM_MESSAGES):
+                time.sleep(SLEEP_TIME_SECS)
+                print(i)
+                producer.send(TOPIC_NAME, 'message' + str(i))
+        except KafkaError:
+            raise Exception('Publish messages unsuccessful')
+        finally:
+            producer.flush()
+            producer.close()
+
+    def consume_messages(self):
+        """Consumes messages using worker 1 as the broker.
+
+          Fails if consuming is unsuccessful or if number of published messages does
+          not match the number of consumed messages.
+
+        Raises:
+          Exception: if consuming results in KafkaError, or mismatch in number of
+            published/consumed messages
+        """
+        node_name = self._get_address(WORKER_1_NODE_SUFFIX, PORT)
+        print('Consuming ' + node_name)
+
+        consumer = KafkaConsumer(
+            TOPIC_NAME,
+            bootstrap_servers=[node_name],
+            auto_offset_reset='smallest',
+            request_timeout_ms=120000)
+
+        try:
+            consumer_iter = iter(consumer)
+
+            # ensures consumer receives all messages sent by producer
+            for _ in range(NUM_MESSAGES):
+                message = consumer_iter.next()
+                print(message.value)
+        except KafkaError:
+            raise Exception('Consume messages unsuccessful')
+        finally:
+            consumer.close()
+
+
+def parse_kafka_config():
+    config = {}
+    with open('/etc/kafka/conf/server.properties') as f:
+        for line in f:
+            line = line.strip()
+            if line and not line.startswith('#'):
+                (key, value) = line.split('=', 1)
+                config[key] = value
+    print(config)
+    return config
+
+
+def verify_kafka_config():
+    config = parse_kafka_config()
+    expected_logs_dirs = sys.argv[1]
+    if config['log.dirs'] != expected_logs_dirs:
+        raise Exception('Invalid log.dirs, expected: {}, actual: {}'.format(
+            expected_logs_dirs, config['log.dirs']))
+
+
+def test_produce_and_consume_messages():
+    cluster_name = get_cluster_name()
+    kafka_runner = KafkaTestRunner(cluster_name)
+    kafka_runner.create_test_topic()
+    kafka_runner.publish_messages()
+    kafka_runner.consume_messages()
+
+
+def main():
+    time.sleep(10)
```

Comment:
Is it to make sure Kafka is up and running? I don't think we should add it in the test, because users won't do the same. If wait is really necessary, it should be in the Kafka init action.